### ISSUE TITLE
关于KD的实现

### ISSUE THREAD

**Issue by Horace-Leon** [zhengli97/CTKD#4]
请问一下，为什么代码实现KD时，教师模型的输出是经过log_softmax，而学生模型输出经过softmax，会有什么影响吗（除了log_softmax计算快）
KD_loss += nn.KLDivLoss(reduction='batchmean')(F.log_softmax(y_s/T, dim=1), F.softmax(y_t/T, dim=1)) * T * T

**Comment by JinYu1998** [comment]
> 请问一下，为什么代码实现KD时，教师模型的输出是经过log_softmax，而学生模型输出经过softmax，会有什么影响吗（除了log_softmax计算快） KD_loss += nn.KLDivLoss(reduction='batchmean')(F.log_softmax(y_s/T, dim=1), F.softmax(y_t/T, dim=1)) * T * T

如果不对教师的输出取log的话, KL可能会变成负的

**Comment by zhengli97** [comment]
在pytorch官网(https://pytorch.org/docs/stable/generated/torch.nn.KLDivLoss.html)里是这么说的
`To avoid underflow issues when computing this quantity, this loss expects the argument input in the log-space.`
pytorch的官网实现里都是这么做的。

**Comment by zhengli97** [comment]
同学如果没有另外的问题，我就关闭这个issue啦。还有想讨论的，欢迎email: zhengli97@qq.com
